tests/support/mod.rs

```diff
@@ -110,13 +110,11 @@ impl SourceConfig for MockSourceConfig {
         let event_counter = self.event_counter.clone();
         let mut recv = wrapped.lock().unwrap().take().unwrap();
         let mut shutdown = Some(shutdown);
-        let mut token = None;
         let source = future::lazy(move || {
             stream::poll_fn(move || {
                 if let Some(until) = shutdown.as_mut() {
                     match until.poll() {
-                        Ok(Async::Ready(res)) => {
-                            token = Some(res);
```

Comment:
Wouldn't this lead to a bug with dropping the token too soon? I suggest adding an explicit `drop` at the end - this might help with letting clippy (and people) know that it must be held on to until the end.

Reply:
Returned back. Wanted to check that CI actually fail if this will be removed. (probably was need create draft PR for this)